tensorflow/python/keras/engine/base_layer.py

```diff
@@ -1868,6 +1868,8 @@ def set_weights(self, weights):
           'expecting %s weights. Provided weights: %s...' %
           (self.name, len(weights), expected_num_weights, str(weights)[:50]))
 
+    weights = [np.array(item) for item in weights]
```

Comment:
Any reason to force convert the into into np.array (which might have non-trival performance overhead)? Also it might hit issue when assigning sparse/ragged variable. 

If this is just trying to avoid ppl from using layer_a.set_weights(layer_b.weights()), shouldn't we just validate the weights here and see if it contains any tf.variable types?

Reply:
That's a good point, I will make the changes and commit again. Thanks for your time.

Reply:
Please don't close this PR, I will commit to this PR again.

Reply:
@shivaditya-meduri Any update on this PR? Please. Thanks!